include/flexflow/ops/noop.h

```diff
@@ -1,38 +1,97 @@
+/**
+ * @file noop.h
+ * @brief NoOp Operator
+ *
+ * @copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 #ifndef _FLEXFLOW_NOOP_H
 #define _FLEXFLOW_NOOP_H
 
 #include "flexflow/model.h"
 
 namespace FlexFlow {
 
+/**
+ * @brief The characteristic of NoOp, combined by a vector of inputs and the
+ * op_type.
+ *
+ * This is actually not a "parameter", but some kind of characteristic to
+ * distinguish different NoOp objects. To maintain the naming consistency, name
+ * this class as "Params".
+ */
+class NoOpParams {
+ public:
+  std::vector<ParallelTensor> inputs;
```

Comment:
I think we should have `.*Params` objects be input-independent, so this field should not be included (happy to discuss this if you disagree).

If this _should_ be included, it almost certainly should be using `ParallelTensorShape` rather than `ParallelTensor`.